gradio/external.py

```diff
@@ -64,23 +67,47 @@ def load(
         )
     if src is None:
         # Separate the repo type (e.g. "model") from repo name (e.g. "google/vit-base-patch16-224")
-        tokens = name.split("/")
-        if len(tokens) <= 1:
+        parts = name.split("/")
+        if len(parts) <= 1:
             raise ValueError(
                 "Either `src` parameter must be provided, or `name` must be formatted as {src}/{repo name}"
             )
-        src = tokens[0]  # type: ignore
-        name = "/".join(tokens[1:])
-    if src in ["huggingface", "models", "spaces"]:
+        src = parts[0]  # type: ignore
+        name = "/".join(parts[1:])
+    assert src is not None  # noqa: S101
+    if not isinstance(src, Callable) and src not in ["models", "spaces", "huggingface"]:
+        raise ValueError(
+            "The `src` parameter must be one of 'huggingface', 'models', 'spaces', or a function that accepts a model name (and optionally, a token), and returns a Gradio app."
+        )
+
+    if not user_provides_token:
+        if isinstance(src, Callable):
+            return src(name, token, **kwargs)
         return load_blocks_from_huggingface(
             name=name, src=src, hf_token=token, **kwargs
         )
-    elif isinstance(src, Callable):
-        return src(name, token, **kwargs)
     else:
-        raise ValueError(
-            "The `src` parameter must be one of 'huggingface', 'models', 'spaces', or a function that accepts a model name (and optionally, a token), and returns a Gradio app."
-        )
+        import gradio as gr
+
+        with gr.Blocks(fill_height=True) as demo:
+            if user_provides_token is True:
+                textbox = gr.Textbox(
+                    type="password",
+                    label="Token",
+                    info="Enter your token and press enter.",
+                )
+            else:
+                textbox = user_provides_token
+
+            @gr.render(inputs=[textbox], triggers=[textbox.submit])
+            def create(token_value):
+                if isinstance(src, Callable):
+                    return src(name, token_value, **kwargs)
+                return load_blocks_from_huggingface(
+                    name=name, src=src, hf_token=token_value, **kwargs
+                )
```

Comment:
I know there is some code duplication, but extracting this out as a separate function and calling it programmatically in `gr.render()` does not work, I tried it here: https://github.com/gradio-app/gradio/pull/9807/commits/b49c64f900e625c1885c457e9502c2cdf65b8472